src/test/fw_c_test_templates/onnxruntime/onnxruntime_c_test.cc

```diff
@@ -0,0 +1,171 @@
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <onnxruntime_c_api.h>
+#include <unistd.h>
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <vector>
+
+#ifdef TRITON_ENABLE_GPU
+#include <cuda_provider_factory.h>
+#include <cuda_runtime_api.h>
+#endif  // TRITON_ENABLE_GPU
+
+const OrtApi* ort_api = OrtGetApiBase()->GetApi(ORT_API_VERSION);
+
+// Check Status of Ort function and exit on error
+void
+CheckStatus(OrtStatus* status)
+{
+  if (status != nullptr) {
+    OrtErrorCode code = ort_api->GetErrorCode(status);
+    const char* msg = ort_api->GetErrorMessage(status);
+    std::cerr << "ONNXRuntime error " << code << ": " << msg;
+    ort_api->ReleaseStatus(status);
+    exit(1);
+  }
+}
+
+void
+Usage(char** argv)
+{
+  std::cerr << "Usage: " << argv[0] << " [options]" << std::endl;
+  std::cerr << "\t-m <model_path>" << std::endl;
+  std::cerr << std::endl;
+
+  exit(1);
+}
+
+int
+main(int argc, char* argv[])
+{
+  std::string model_path = "model.onnx";
+
+  // Parse commandline...
+  int opt;
+  while ((opt = getopt(argc, argv, "m:")) != -1) {
+    switch (opt) {
+      case 'm':
+        model_path = optarg;
+        break;
+      case '?':
+        Usage(argv);
+        break;
+    }
+  }
+
+  // Initialize  enviroment
+  OrtEnv* env;
+  CheckStatus(ort_api->CreateEnv(ORT_LOGGING_LEVEL_WARNING, "test", &env));
+
+  // Initialize session options
+  OrtSessionOptions* session_options;
+  CheckStatus(ort_api->CreateSessionOptions(&session_options));
+  CheckStatus(ort_api->SetIntraOpNumThreads(session_options, 1));
+
+  // Set graph optimization level
+  GraphOptimizationLevel optimization_level =
+      GraphOptimizationLevel::ORT_ENABLE_ALL;
+  CheckStatus(ort_api->SetSessionGraphOptimizationLevel(
+      session_options, optimization_level));
+
+  // Optionally add more execution providers via session_options
+#ifdef TRITON_ENABLE_GPU
+  size_t gpu_device_id = 0;
+  CheckStatus(OrtSessionOptionsAppendExecutionProvider_CUDA(
+      session_options, gpu_device_id));
+#endif  // TRITON_ENABLE_GPU
+
+  printf("Test using native Onnxruntime C API\n");
+  OrtSession* session;
+  CheckStatus(ort_api->CreateSession(
+      env, model_path.c_str(), session_options, &session));
+
+  OrtAllocator* allocator;
+  CheckStatus(ort_api->GetAllocatorWithDefaultOptions(&allocator));
+  const OrtMemoryInfo* allocator_info;
+  CheckStatus(ort_api->AllocatorGetInfo(allocator, &allocator_info));
+
+  // This model has only 2 inputs and 1 output node.
+  // input__0: {4096, 13} FP16
+  // input__1: {4096, 26} INT64
+  // output__0: {4096, 1} FP16
+
+  // Use OrtSession to get metadata of inputs and outputs if not known
+  std::vector<const char*> input_names = {"input__0", "input__1"};
```

Comment:
Shouldn't we derive these parameters directly from the ONNX model itself?

Reply:
This is just a template. I could make it generic and allow it to run for any kind of model but for now I've only designed it for the DLRM model.